internal/services/gitserver/repo-cleaner.go

```diff
@@ -0,0 +1,197 @@
+package gitserver
+
+import (
+	"context"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"agola.io/agola/internal/services/config"
+	"agola.io/agola/internal/util"
+	"go.uber.org/zap"
+)
+
+type RepoCleaner struct {
+	dataDir          string
+	logger           *zap.SugaredLogger
+	scanTime         time.Duration
+	deleteBranchTime time.Duration
+}
+
+const (
+	DEFAULT_REPO_DELETE_BRANCH string = "720h"
+	DEFAULT_SCAN_TIME          string = "24h"
+)
+
+func NewRepoCleaner(c *config.Gitserver, l *zap.SugaredLogger) *RepoCleaner {
+	stDuration, err := time.ParseDuration(c.RepoCleaner.ScanTime)
+	if err != nil {
+		l.Error("repo-cleaner error:", err)
+		stDuration, _ = time.ParseDuration(DEFAULT_SCAN_TIME)
+	}
+
+	btDuration, err := time.ParseDuration(c.RepoCleaner.DeleteBranchAge)
+	if err != nil {
+		l.Error("repo-cleaner error:", err)
+		btDuration, _ = time.ParseDuration(DEFAULT_REPO_DELETE_BRANCH)
```

Comment:
All these time.ParseDuration aren't really needed. Just make the constant of type time.Duration (see previous comment)